### PR TITLE
NumpyAwareEncoder Class Monkey Patch

### PR DESCRIPTION
**Pull request by ThomasSchellenbergNextCentury**
Added NumpyAwareEncoder override class to fix an issue with passing numpy arrays to Unity.

@bzinberg @fplk Can you please verify that this branch fixes the numpy array issue for you? I'll also add some new tests as part of my current [integration testing](https://github.com/NextCenturyCorporation/MCS/pull/225) ticket.

### CODE REVIEW

**Review by brianpippin — APPROVED** [review]

File: machine_common_sense/controller.py [hunk 1]
```diff
@@ -56,6 +54,19 @@ def __image_depth_override(self, image_depth_data, **kwargs):
 ai2thor.server.Event._image_depth = __image_depth_override
 
 
+class NumpyAwareEncoderOverride(json.JSONEncoder):
+    # From https://github.com/allenai/ai2thor/blob/master/ai2thor/server.py#L17-L24 # noqa: E501
```
Comment (bzinberg) [inline machine_common_sense/controller.py:58]: Suggest changing this to a permalink so it doesn't go stale if the upstream repo changes
```suggestion
    # From https://github.com/allenai/ai2thor/blob/bd35d2cb887faee8b87aa04bd9373b027eb39f17/ai2thor/server.py#L17-L24 # noqa: E501
```

**Review by bzinberg — COMMENTED** [review]
Looks good, thanks @ThomasSchellenbergNextCentury!  One tiny suggestion for now; next, I'm about to test it on my machine and will get back to you.

**Comment by bzinberg** [comment]
I am re-running the scene `beta_0002_01_debug.json` which @fplk reported a [crash](https://machinecommonsense.slack.com/archives/CTR31V12M/p1611636989043700?thread_ts=1611632022.042000&cid=CTR31V12M) on and I [reproduced](https://machinecommonsense.slack.com/archives/CTR31V12M/p1611685170048700?thread_ts=1611632022.042000&cid=CTR31V12M) at the time.  For the re-run now, I am using the current version of our TA1 team's infra, v0.3.7 of the Unity app, and this branch of the MCS python library.

I'm getting the following, different error:
```
┌ Warning: Skipping scene: run_scene() encountered exception: BoundsError: attempt to access 0-element Array{Any,1} at index [0]
│ Stacktrace:
│  [1] getindex at ./array.jl:809 [inlined]
│  [2] last(::Array{Any,1}) at ./abstractarray.jl:362
│  [3] extract_transform_params_from_initial_frame_agency(::PyCall.PyObject, ::GenGridSLAM.CameraIntrinsics; resolution::Float64) at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgency.jl/src/perception.jl:13
│  [4] extract_transform_params_from_initial_frame_agency at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgency.jl/src/perception.jl:13 [inlined]
│  [5] solve_agency_online(::PyCall.PyObject, ::McsAgent, ::McsConfiguration, ::PyCall.PyObject, ::Dict{String,String}; tv_thresh::Float64, ll_thresh::Float64, tv_step_thresh::Float64, ll_step_thresh::Float64, n_samples::Int64, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgency.jl/src/GenAgency.jl:151
│  [6] solve_agency_online at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgency.jl/src/GenAgency.jl:151 [inlined]
│  [7] #solve_agency#92 at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgency.jl/src/GenAgency.jl:264 [inlined]
│  [8] solve_agency at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgency.jl/src/GenAgency.jl:263 [inlined]
│  [9] run_scene(::PyCall.PyObject, ::Function, ::McsConfiguration, ::String, ::Array{Float64,2}) at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgent.jl/src/GenAgent.jl:87
│  [10] macro expansion at ./timing.jl:233 [inlined]
│  [11] run(::Array{String,1}, ::String, ::Bool, ::Bool) at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgent.jl/src/GenAgent.jl:236
│  [12] run(::Array{String,1}, ::String) at /home/bzinberg/probcomp/GenPRAM_workspace/GenAgent.jl/src/GenAgent.jl:220
│  [13] top-level scope at none:6
│  [14] eval(::Module, ::Any) at ./boot.jl:331
│  [15] exec_options(::Base.JLOptions) at ./client.jl:272
│  [16] _start() at ./client.jl:506
└ @ GenAgent ~/probcomp/GenPRAM_workspace/GenAgent.jl/src/GenAgent.jl:171
```
This might be an issue with our infra, so ball is in my or @fplk's court to diagnose and get back to you.

**Comment by bzinberg** [comment]
Ah, it was because part of our codebase was still looking for an `mcs_config.yaml` to determine the metadata level, but as of MCS 0.3.7 that's no longer what the simulator uses.  Fixing and then I'll retry.

**Review by bzinberg — APPROVED** [review]
@ThomasSchellenbergNextCentury, it works on my end!  I think we're ready to merge.

@fplk, the stop-gap changes I made to the way GenAgent.jl interacts with the simulator are https://github.com/probcomp/GenAgent.jl/pull/68.  We should update the metadata level plumbing code to do this configurably.  (Thomas, LMK if you want access to PRs like this, if so we can buy a collaborator seat for you in our GitHub org.)

**Pull request merged**